docs/DafnyRef/UserGuide.md

```diff
@@ -365,7 +365,21 @@ The command emits exit codes of
 
 #### 25.5.1.9. `dafny format` {#sec-dafny-format}
 
-This command is not yet released, but will be a command that formats source code to a consistent style.
+Dafny supports a formatter, which for now only changes the indentation of every line in a Dafny file, so that it conforms
+to the idiomatic Dafny code formatting style.
+For the formatter to work, the file should be parsed correctly by Dafny.
+
+There are four ways to use the formatter:
+
+* `dafny format file.dfy [file2.dfy ...]` formats the given Dafny files, altering the files in place.
+* `dafny format folder` formats all the Dafny files contain in the given folders, altering the files in place. For example, `dafny format .` formats all the Dafny files recursively in the current folder.
+* `dafny format --print file.dfy` formats each file but instead of altering the files, output the formatted content to stdout
+* `dafny format --check file.dfy` does not alter files. It will print a message concerning which files need formatting.
+
+You can also use `--stdin` instead of providing a file to format a full Dafny file from the standard input.
+
+In any case, each version of `dafny format` returns a non-zero return code
```

Comment:
Which error code is being used to indicate unformatted input? Something different than other error codes I hope.
I would have suggested a non-zero error code only on
- command-line errors (1, as for dafny)
- parse/resolution errors (2, as for dafny)
- unformatted input, but only for --check -- perhaps 5, to be different than other kinds of errors

Reply:
I'm using `ExitValue.DAFNY_ERROR`, which is 2. Switched to a new value (probably 5), and only if --check is present.